rfcs/0009-separate-wal.md

```diff
@@ -0,0 +1,99 @@
+# Separate Object Store for WAL
+
+Status: Draft
+
+Authors:
+
+* [Sergei Sitnikov](https://github.com/taburet)
+
+## Motivation
+
+Sequential writes are a common pattern. Combined with `await_durable` set to
+`true`, which is the default, WAL performance is the main limiting factor for
+the sequential writes throughput. Even if `flush_interval` is set to an extremely
+low value, it can't overcome the inherent latency of the underlying object store.
+
+The obvious solution is switching to a lower latency object store; for instance,
+from S3 to S3 Express One Zone. But lower latency stores usually come with a cost,
+they might be significantly more expensive or might be limited in features. Using
+such stores for the main store containing the entire database might be undesirable.
+
+## Goals
```

Comment:
I like this RFC! It's a nice simple way to improve the WAL performance without dramatically changing the code to support other storage formats.

I'm curious whether this could also help with keeping read replicas up to date, if you have a WAL on S3 Express One Zone you could hypothetically actively read from the WAL into a temporary memtable that you discard whenever you read a new (leveled) SST from S3.

Reply:
💯 I totally think it could. The main win is you can poll more frequently since reads are cheaper on S3E1Z.